watchdog/watchdog.go

```diff
@@ -10,34 +10,29 @@ import (
 	"github.com/TwinProduction/gatus/config"
 	"github.com/TwinProduction/gatus/core"
 	"github.com/TwinProduction/gatus/metric"
+	"github.com/TwinProduction/gatus/storage"
 )
 
 var (
-	serviceStatuses = make(map[string]*core.ServiceStatus)
-
-	// serviceStatusesMutex is used to prevent concurrent map access
-	serviceStatusesMutex sync.RWMutex
+	store = storage.NewInMemoryStore()
 
 	// monitoringMutex is used to prevent multiple services from being evaluated at the same time.
 	// Without this, conditions using response time may become inaccurate.
 	monitoringMutex sync.Mutex
 )
 
 // GetJSONEncodedServiceStatuses returns a list of core.ServiceStatus for each services encoded using json.Marshal.
-// The reason why the encoding is done here is because we use a mutex to prevent concurrent map access.
 func GetJSONEncodedServiceStatuses() ([]byte, error) {
```

Comment:
Could you rename that function to `GetServiceStatusesAsJSON`?

I don't know why I called it like that - JSON's not even an encoding 🤦‍♂️

Reply:
5eb289c